src/RawRabbit/Operations/Publisher.cs

```diff
@@ -57,15 +56,19 @@ public Task PublishAsync<TMessage>(TMessage message, Guid globalMessageId, Publi
 					{
 						throw t.Exception;
 					}
+
+					channelTask.Result.BasicReturn += config.BasicReturn;
```

Comment:
The channel in the `channelTask` is reused when publishing messages. I think we need to remove the event handler once the message is delivered (which is when the `ackTask` completes). One way to do this is to use `ContinueWith` on the `ackTask` and remove it from there. What do you think?